mailet/standard/src/main/java/org/apache/james/transport/mailets/ContactExtractor.java

```diff
@@ -95,26 +101,35 @@ public void service(Mail mail) throws MessagingException {
         }
     }
 
-    private Optional<String> extractContacts(Mail mail) throws MessagingException, IOException {
-        MimeMessage message = mail.getMessage();
+    @VisibleForTesting
+    Optional<String> extractContacts(Mail mail) throws MessagingException, IOException {
+        ImmutableList<String> allRecipients = getAllRecipients(mail.getMessage());
 
         return Optional.of(mail.getSender())
             .map(MailAddress::asString)
-            .filter(Throwing.predicate(sender -> hasRecipients(message)))
-            .map(Throwing.function(sender -> new ExtractedContacts(sender, recipients(message))))
+            .filter(sender -> !allRecipients.isEmpty())
```

Comment:
you don't use sender at all ?

Reply:
not in the filter, but we construct the response with it. I don't see what I can change about this.

Reply:
using a `if` ?

Reply:
I agree with @mbaechler filter when you don't use the inside of the Optional is cumbersome